// @alicloud/dingtalk/dist/mail_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMailFoldersResponseBodyFolders = exports.ListMailFoldersResponse = exports.ListMailFoldersResponseBody = exports.ListMailFoldersRequest = exports.ListMailFoldersHeaders = exports.CreateUserResponse = exports.CreateUserResponseBody = exports.CreateUserRequest = exports.CreateUserHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CreateUserHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateUserHeaders = CreateUserHeaders;
class CreateUserRequest extends $tea.Model {
    static names() {
        return {
            email: 'email',
            employeeType: 'employeeType',
            name: 'name',
            password: 'password',
        };
    }
    static types() {
        return {
            email: 'string',
            employeeType: 'string',
            name: 'string',
            password: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateUserRequest = CreateUserRequest;
class CreateUserResponseBody extends $tea.Model {
    static names() {
        return {
            email: 'email',
        };
    }
    static types() {
        return {
            email: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateUserResponseBody = CreateUserResponseBody;
class CreateUserResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateUserResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateUserResponse = CreateUserResponse;
class ListMailFoldersHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListMailFoldersHeaders = ListMailFoldersHeaders;
class ListMailFoldersRequest extends $tea.Model {
    static names() {
        return {
            folderId: 'folderId',
        };
    }
    static types() {
        return {
            folderId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListMailFoldersRequest = ListMailFoldersRequest;
class ListMailFoldersResponseBody extends $tea.Model {
    static names() {
        return {
            folders: 'folders',
        };
    }
    static types() {
        return {
            folders: { 'type': 'array', 'itemType': ListMailFoldersResponseBodyFolders },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListMailFoldersResponseBody = ListMailFoldersResponseBody;
class ListMailFoldersResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListMailFoldersResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListMailFoldersResponse = ListMailFoldersResponse;
class ListMailFoldersResponseBodyFolders extends $tea.Model {
    static names() {
        return {
            childFolderCount: 'childFolderCount',
            displayName: 'displayName',
            extensions: 'extensions',
            id: 'id',
            parentFolderId: 'parentFolderId',
            totalItemCount: 'totalItemCount',
            unreadItemCount: 'unreadItemCount',
        };
    }
    static types() {
        return {
            childFolderCount: 'number',
            displayName: 'string',
            extensions: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            id: 'string',
            parentFolderId: 'string',
            totalItemCount: 'number',
            unreadItemCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ListMailFoldersResponseBodyFolders = ListMailFoldersResponseBodyFolders;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        let gatewayClient = new gateway_dingtalk_1.default();
        this._spi = gatewayClient;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * 创建企业邮箱用户
     *
     * @param request - CreateUserRequest
     * @param headers - CreateUserHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns CreateUserResponse
     */
    async createUserWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.email)) {
            body["email"] = request.email;
        }
        if (!tea_util_1.default.isUnset(request.employeeType)) {
            body["employeeType"] = request.employeeType;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.password)) {
            body["password"] = request.password;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateUser",
            version: "mail_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/mail/users`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateUserResponse({}));
    }
    /**
     * 创建企业邮箱用户
     *
     * @param request - CreateUserRequest
     * @returns CreateUserResponse
     */
    async createUser(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateUserHeaders({});
        return await this.createUserWithOptions(request, headers, runtime);
    }
    /**
     * 获取指定文件夹的子文件夹列表
     *
     * @param request - ListMailFoldersRequest
     * @param headers - ListMailFoldersHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns ListMailFoldersResponse
     */
    async listMailFoldersWithOptions(email, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.folderId)) {
            query["folderId"] = request.folderId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListMailFolders",
            version: "mail_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/mail/users/${email}/mailFolders`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListMailFoldersResponse({}));
    }
    /**
     * 获取指定文件夹的子文件夹列表
     *
     * @param request - ListMailFoldersRequest
     * @returns ListMailFoldersResponse
     */
    async listMailFolders(email, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListMailFoldersHeaders({});
        return await this.listMailFoldersWithOptions(email, request, headers, runtime);
    }
}
exports.default = Client;
