// @alicloud/dingtalk/dist/bay_max_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryBaymaxSkillLogResponse = exports.QueryBaymaxSkillLogResponseBody = exports.QueryBaymaxSkillLogRequest = exports.QueryBaymaxSkillLogHeaders = void 0;
// This file is auto-generated, don't edit it
/**
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class QueryBaymaxSkillLogHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBaymaxSkillLogHeaders = QueryBaymaxSkillLogHeaders;
class QueryBaymaxSkillLogRequest extends $tea.Model {
    static names() {
        return {
            from: 'from',
            skillExecuteId: 'skillExecuteId',
            to: 'to',
        };
    }
    static types() {
        return {
            from: 'number',
            skillExecuteId: 'string',
            to: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBaymaxSkillLogRequest = QueryBaymaxSkillLogRequest;
class QueryBaymaxSkillLogResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBaymaxSkillLogResponseBody = QueryBaymaxSkillLogResponseBody;
class QueryBaymaxSkillLogResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryBaymaxSkillLogResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryBaymaxSkillLogResponse = QueryBaymaxSkillLogResponse;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        let gatewayClient = new gateway_dingtalk_1.default();
        this._spi = gatewayClient;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * Baymax技能执行日志
     *
     * @param request - QueryBaymaxSkillLogRequest
     * @param headers - QueryBaymaxSkillLogHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns QueryBaymaxSkillLogResponse
     */
    async queryBaymaxSkillLogWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.from)) {
            query["from"] = request.from;
        }
        if (!tea_util_1.default.isUnset(request.skillExecuteId)) {
            query["skillExecuteId"] = request.skillExecuteId;
        }
        if (!tea_util_1.default.isUnset(request.to)) {
            query["to"] = request.to;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryBaymaxSkillLog",
            version: "bayMax_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/bayMax/skills/logs`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryBaymaxSkillLogResponse({}));
    }
    /**
     * Baymax技能执行日志
     *
     * @param request - QueryBaymaxSkillLogRequest
     * @returns QueryBaymaxSkillLogResponse
     */
    async queryBaymaxSkillLog(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryBaymaxSkillLogHeaders({});
        return await this.queryBaymaxSkillLogWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
