// @alicloud/dingtalk/src/vip_member_1_0/client.ts
// This file is auto-generated, don't edit it
/**
 */
import Util, * as $Util from '@alicloud/tea-util';
import GatewayClient from '@alicloud/gateway-dingtalk';
import OpenApi, * as $OpenApi from '@alicloud/openapi-client';
import OpenApiUtil from '@alicloud/openapi-util';
import * as $tea from '@alicloud/tea-typescript';

export class QueryVipMemberInfoHeaders extends $tea.Model {
  commonHeaders?: { [key: string]: string };
  xAcsDingtalkAccessToken?: string;
  static names(): { [key: string]: string } {
    return {
      commonHeaders: 'commonHeaders',
      xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
    };
  }

  static types(): { [key: string]: any } {
    return {
      commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
      xAcsDingtalkAccessToken: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}

export class QueryVipMemberInfoRequest extends $tea.Model {
  channelCode?: string;
  static names(): { [key: string]: string } {
    return {
      channelCode: 'channelCode',
    };
  }

  static types(): { [key: string]: any } {
    return {
      channelCode: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}

export class QueryVipMemberInfoResponseBody extends $tea.Model {
  expireTime?: string;
  isVip?: boolean;
  static names(): { [key: string]: string } {
    return {
      expireTime: 'expireTime',
      isVip: 'isVip',
    };
  }

  static types(): { [key: string]: any } {
    return {
      expireTime: 'string',
      isVip: 'boolean',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}

export class QueryVipMemberInfoResponse extends $tea.Model {
  headers?: { [key: string]: string };
  statusCode?: number;
  body?: QueryVipMemberInfoResponseBody;
  static names(): { [key: string]: string } {
    return {
      headers: 'headers',
      statusCode: 'statusCode',
      body: 'body',
    };
  }

  static types(): { [key: string]: any } {
    return {
      headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
      statusCode: 'number',
      body: QueryVipMemberInfoResponseBody,
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}


export default class Client extends OpenApi {

  constructor(config: $OpenApi.Config) {
    super(config);
    let gatewayClient = new GatewayClient();
    this._spi = gatewayClient;
    this._endpointRule = "";
    if (Util.empty(this._endpoint)) {
      this._endpoint = "api.dingtalk.com";
    }

  }


  /**
   * 查询365会员信息
   * 
   * @param request - QueryVipMemberInfoRequest
   * @param headers - QueryVipMemberInfoHeaders
   * @param runtime - runtime options for this request RuntimeOptions
   * @returns QueryVipMemberInfoResponse
   */
  async queryVipMemberInfoWithOptions(request: QueryVipMemberInfoRequest, headers: QueryVipMemberInfoHeaders, runtime: $Util.RuntimeOptions): Promise<QueryVipMemberInfoResponse> {
    Util.validateModel(request);
    let body : {[key: string ]: any} = { };
    if (!Util.isUnset(request.channelCode)) {
      body["channelCode"] = request.channelCode;
    }

    let realHeaders : {[key: string ]: string} = { };
    if (!Util.isUnset(headers.commonHeaders)) {
      realHeaders = headers.commonHeaders;
    }

    if (!Util.isUnset(headers.xAcsDingtalkAccessToken)) {
      realHeaders["x-acs-dingtalk-access-token"] = Util.toJSONString(headers.xAcsDingtalkAccessToken);
    }

    let req = new $OpenApi.OpenApiRequest({
      headers: realHeaders,
      body: OpenApiUtil.parseToMap(body),
    });
    let params = new $OpenApi.Params({
      action: "QueryVipMemberInfo",
      version: "vipMember_1.0",
      protocol: "HTTP",
      pathname: `/v1.0/vipMember/users/memberInfos/query`,
      method: "POST",
      authType: "AK",
      style: "ROA",
      reqBodyType: "none",
      bodyType: "json",
    });
    return $tea.cast<QueryVipMemberInfoResponse>(await this.execute(params, req, runtime), new QueryVipMemberInfoResponse({}));
  }

  /**
   * 查询365会员信息
   * 
   * @param request - QueryVipMemberInfoRequest
   * @returns QueryVipMemberInfoResponse
   */
  async queryVipMemberInfo(request: QueryVipMemberInfoRequest): Promise<QueryVipMemberInfoResponse> {
    let runtime = new $Util.RuntimeOptions({ });
    let headers = new QueryVipMemberInfoHeaders({ });
    return await this.queryVipMemberInfoWithOptions(request, headers, runtime);
  }

}
