// @alicloud/dingtalk/dist/live_1_0/client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetOrgLiveListHeaders = exports.GetLiveReplayUrlResponse = exports.GetLiveReplayUrlResponseBody = exports.GetLiveReplayUrlRequest = exports.GetLiveReplayUrlHeaders = exports.GetGroupLiveListResponse = exports.GetGroupLiveListResponseBody = exports.GetGroupLiveListShrinkRequest = exports.GetGroupLiveListRequest = exports.GetGroupLiveListHeaders = exports.EditFeedReplayResponse = exports.EditFeedReplayResponseBody = exports.EditFeedReplayRequest = exports.EditFeedReplayHeaders = exports.DeleteLiveFeedResponse = exports.DeleteLiveFeedResponseBody = exports.DeleteLiveFeedRequest = exports.DeleteLiveFeedHeaders = exports.DeleteLiveResponse = exports.DeleteLiveResponseBody = exports.DeleteLiveRequest = exports.DeleteLiveHeaders = exports.DelLiveNoticeWidgetResponse = exports.DelLiveNoticeWidgetResponseBody = exports.DelLiveNoticeWidgetRequest = exports.DelLiveNoticeWidgetHeaders = exports.DelLiveInteractionPluginResponse = exports.DelLiveInteractionPluginResponseBody = exports.DelLiveInteractionPluginRequest = exports.DelLiveInteractionPluginHeaders = exports.CreateLiveResponse = exports.CreateLiveResponseBody = exports.CreateLiveRequest = exports.CreateLiveHeaders = exports.CreateCloudFeedResponse = exports.CreateCloudFeedResponseBody = exports.CreateCloudFeedRequest = exports.CreateCloudFeedHeaders = exports.AddShareCidListResponse = exports.AddShareCidListResponseBody = exports.AddShareCidListRequest = exports.AddShareCidListHeaders = exports.AddLiveNoticeWidgetResponse = exports.AddLiveNoticeWidgetResponseBody = exports.AddLiveNoticeWidgetRequest = exports.AddLiveNoticeWidgetHeaders = exports.AddLiveInteractionPluginResponse = exports.AddLiveInteractionPluginResponseBody = exports.AddLiveInteractionPluginRequest = exports.AddLiveInteractionPluginHeaders = void 0;
exports.SendLiveInteractionPluginEffectsMsgResponse = exports.SendLiveInteractionPluginEffectsMsgResponseBody = exports.SendLiveInteractionPluginEffectsMsgRequest = exports.SendLiveInteractionPluginEffectsMsgHeaders = exports.QuerySubscribeStatusResponse = exports.QuerySubscribeStatusResponseBody = exports.QuerySubscribeStatusShrinkRequest = exports.QuerySubscribeStatusRequest = exports.QuerySubscribeStatusHeaders = exports.QueryLiveWatchUserListResponse = exports.QueryLiveWatchUserListResponseBody = exports.QueryLiveWatchUserListRequest = exports.QueryLiveWatchUserListHeaders = exports.QueryLiveWatchDetailResponse = exports.QueryLiveWatchDetailResponseBody = exports.QueryLiveWatchDetailRequest = exports.QueryLiveWatchDetailHeaders = exports.QueryLiveInteractionPluginResponse = exports.QueryLiveInteractionPluginResponseBody = exports.QueryLiveInteractionPluginRequest = exports.QueryLiveInteractionPluginHeaders = exports.QueryLiveInfoResponse = exports.QueryLiveInfoResponseBody = exports.QueryLiveInfoRequest = exports.QueryLiveInfoHeaders = exports.QueryFeedWhiteListResponse = exports.QueryFeedWhiteListResponseBody = exports.QueryFeedWhiteListRequest = exports.QueryFeedWhiteListHeaders = exports.ModifyFeedWhiteListResponse = exports.ModifyFeedWhiteListResponseBody = exports.ModifyFeedWhiteListShrinkRequest = exports.ModifyFeedWhiteListRequest = exports.ModifyFeedWhiteListHeaders = exports.GetUserWatchLiveListResponse = exports.GetUserWatchLiveListResponseBody = exports.GetUserWatchLiveListRequest = exports.GetUserWatchLiveListHeaders = exports.GetUserCreateLiveListResponse = exports.GetUserCreateLiveListResponseBody = exports.GetUserCreateLiveListRequest = exports.GetUserCreateLiveListHeaders = exports.GetUserAllLiveListResponse = exports.GetUserAllLiveListResponseBody = exports.GetUserAllLiveListRequest = exports.GetUserAllLiveListHeaders = exports.GetOrgLiveListResponse = exports.GetOrgLiveListResponseBody = exports.GetOrgLiveListShrinkRequest = exports.GetOrgLiveListRequest = void 0;
exports.GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListHasSubscribed = exports.GetUserAllLiveListResponseBodyResult = exports.GetUserAllLiveListResponseBodyResultLiveInfoPopModelList = exports.GetUserAllLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo = exports.GetUserAllLiveListResponseBodyResultLiveInfoPopModelListExtraInfo = exports.GetOrgLiveListResponseBodyResult = exports.GetOrgLiveListResponseBodyResultUpdateLive = exports.GetOrgLiveListResponseBodyResultUpdateLiveLiveList = exports.GetOrgLiveListResponseBodyResultNewLive = exports.GetOrgLiveListResponseBodyResultNewLiveLiveList = exports.GetOrgLiveListRequestRequestBody = exports.GetLiveReplayUrlResponseBodyResult = exports.GetGroupLiveListResponseBodyResult = exports.GetGroupLiveListResponseBodyResultGroupLiveList = exports.GetGroupLiveListRequestRequestBody = exports.DeleteLiveResponseBodyResult = exports.DelLiveNoticeWidgetResponseBodyResult = exports.DelLiveInteractionPluginResponseBodyResult = exports.CreateLiveResponseBodyResult = exports.AddLiveNoticeWidgetResponseBodyResult = exports.AddLiveInteractionPluginResponseBodyResult = exports.UpdateLiveInteractionPluginResponse = exports.UpdateLiveInteractionPluginResponseBody = exports.UpdateLiveInteractionPluginRequest = exports.UpdateLiveInteractionPluginHeaders = exports.UpdateLiveFeedResponse = exports.UpdateLiveFeedResponseBody = exports.UpdateLiveFeedRequest = exports.UpdateLiveFeedHeaders = exports.UpdateLiveResponse = exports.UpdateLiveResponseBody = exports.UpdateLiveRequest = exports.UpdateLiveHeaders = exports.SubscribeLiveResponse = exports.SubscribeLiveResponseBody = exports.SubscribeLiveRequest = exports.SubscribeLiveHeaders = exports.StopCloudFeedResponse = exports.StopCloudFeedResponseBody = exports.StopCloudFeedRequest = exports.StopCloudFeedHeaders = exports.StartCloudFeedResponse = exports.StartCloudFeedResponseBody = exports.StartCloudFeedRequest = exports.StartCloudFeedHeaders = exports.SendLivePluginUserActionMsgResponse = exports.SendLivePluginUserActionMsgResponseBody = exports.SendLivePluginUserActionMsgShrinkRequest = exports.SendLivePluginUserActionMsgRequest = exports.SendLivePluginUserActionMsgHeaders = void 0;
exports.UpdateLiveInteractionPluginResponseBodyResult = exports.UpdateLiveResponseBodyResult = exports.SubscribeLiveResponseBodyResult = exports.SendLivePluginUserActionMsgRequestPluginEffectsMessage = exports.SendLiveInteractionPluginEffectsMsgResponseBodyResult = exports.QuerySubscribeStatusResponseBodyResult = exports.QuerySubscribeStatusResponseBodyResultSubscribeStatusDTOS = exports.QuerySubscribeStatusRequestBody = exports.QueryLiveWatchUserListResponseBodyResult = exports.QueryLiveWatchUserListResponseBodyResultOutOrgUserList = exports.QueryLiveWatchUserListResponseBodyResultOrgUsesList = exports.QueryLiveWatchDetailResponseBodyResult = exports.QueryLiveInteractionPluginResponseBodyResult = exports.QueryLiveInteractionPluginResponseBodyResultLivePluginList = exports.QueryLiveInfoResponseBodyResult = exports.QueryLiveInfoResponseBodyResultLiveInfo = exports.GetUserWatchLiveListResponseBodyResult = exports.GetUserWatchLiveListResponseBodyResultLiveInfoPopModelList = exports.GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo = exports.GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListExtraInfo = exports.GetUserCreateLiveListResponseBodyResult = exports.GetUserCreateLiveListResponseBodyResultLiveInfoPopModelList = exports.GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo = void 0;
// This file is auto-generated, don't edit it
/**
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddLiveInteractionPluginHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveInteractionPluginHeaders = AddLiveInteractionPluginHeaders;
class AddLiveInteractionPluginRequest extends $tea.Model {
    static names() {
        return {
            anchorJumpUrl: 'anchorJumpUrl',
            pluginIconUrl: 'pluginIconUrl',
            pluginName: 'pluginName',
            pluginNameEn: 'pluginNameEn',
            viewerJumpUrl: 'viewerJumpUrl',
            liveId: 'liveId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            anchorJumpUrl: 'string',
            pluginIconUrl: 'string',
            pluginName: 'string',
            pluginNameEn: 'string',
            viewerJumpUrl: 'string',
            liveId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveInteractionPluginRequest = AddLiveInteractionPluginRequest;
class AddLiveInteractionPluginResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: AddLiveInteractionPluginResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveInteractionPluginResponseBody = AddLiveInteractionPluginResponseBody;
class AddLiveInteractionPluginResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddLiveInteractionPluginResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveInteractionPluginResponse = AddLiveInteractionPluginResponse;
class AddLiveNoticeWidgetHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveNoticeWidgetHeaders = AddLiveNoticeWidgetHeaders;
class AddLiveNoticeWidgetRequest extends $tea.Model {
    static names() {
        return {
            iconUrl: 'iconUrl',
            jumpUrl: 'jumpUrl',
            liveId: 'liveId',
            noticeText: 'noticeText',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            iconUrl: 'string',
            jumpUrl: 'string',
            liveId: 'string',
            noticeText: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveNoticeWidgetRequest = AddLiveNoticeWidgetRequest;
class AddLiveNoticeWidgetResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: AddLiveNoticeWidgetResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveNoticeWidgetResponseBody = AddLiveNoticeWidgetResponseBody;
class AddLiveNoticeWidgetResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddLiveNoticeWidgetResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveNoticeWidgetResponse = AddLiveNoticeWidgetResponse;
class AddShareCidListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddShareCidListHeaders = AddShareCidListHeaders;
class AddShareCidListRequest extends $tea.Model {
    static names() {
        return {
            groupIdType: 'groupIdType',
            groupIds: 'groupIds',
            userId: 'userId',
        };
    }
    static types() {
        return {
            groupIdType: 'number',
            groupIds: { 'type': 'array', 'itemType': 'string' },
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddShareCidListRequest = AddShareCidListRequest;
class AddShareCidListResponseBody extends $tea.Model {
    static names() {
        return {
            hasShareSuccess: 'hasShareSuccess',
            shareSuccessGroupList: 'shareSuccessGroupList',
        };
    }
    static types() {
        return {
            hasShareSuccess: 'boolean',
            shareSuccessGroupList: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddShareCidListResponseBody = AddShareCidListResponseBody;
class AddShareCidListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddShareCidListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddShareCidListResponse = AddShareCidListResponse;
class CreateCloudFeedHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateCloudFeedHeaders = CreateCloudFeedHeaders;
class CreateCloudFeedRequest extends $tea.Model {
    static names() {
        return {
            coverUrl: 'coverUrl',
            intro: 'intro',
            startTime: 'startTime',
            title: 'title',
            userId: 'userId',
            videoUrl: 'videoUrl',
        };
    }
    static types() {
        return {
            coverUrl: 'string',
            intro: 'string',
            startTime: 'number',
            title: 'string',
            userId: 'string',
            videoUrl: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateCloudFeedRequest = CreateCloudFeedRequest;
class CreateCloudFeedResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateCloudFeedResponseBody = CreateCloudFeedResponseBody;
class CreateCloudFeedResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateCloudFeedResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateCloudFeedResponse = CreateCloudFeedResponse;
class CreateLiveHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateLiveHeaders = CreateLiveHeaders;
class CreateLiveRequest extends $tea.Model {
    static names() {
        return {
            coverUrl: 'coverUrl',
            introduction: 'introduction',
            preEndTime: 'preEndTime',
            preStartTime: 'preStartTime',
            publicType: 'publicType',
            title: 'title',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            coverUrl: 'string',
            introduction: 'string',
            preEndTime: 'number',
            preStartTime: 'number',
            publicType: 'number',
            title: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateLiveRequest = CreateLiveRequest;
class CreateLiveResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: CreateLiveResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateLiveResponseBody = CreateLiveResponseBody;
class CreateLiveResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateLiveResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateLiveResponse = CreateLiveResponse;
class DelLiveInteractionPluginHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveInteractionPluginHeaders = DelLiveInteractionPluginHeaders;
class DelLiveInteractionPluginRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            pluginId: 'pluginId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            pluginId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveInteractionPluginRequest = DelLiveInteractionPluginRequest;
class DelLiveInteractionPluginResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: DelLiveInteractionPluginResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveInteractionPluginResponseBody = DelLiveInteractionPluginResponseBody;
class DelLiveInteractionPluginResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DelLiveInteractionPluginResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveInteractionPluginResponse = DelLiveInteractionPluginResponse;
class DelLiveNoticeWidgetHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveNoticeWidgetHeaders = DelLiveNoticeWidgetHeaders;
class DelLiveNoticeWidgetRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveNoticeWidgetRequest = DelLiveNoticeWidgetRequest;
class DelLiveNoticeWidgetResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: DelLiveNoticeWidgetResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveNoticeWidgetResponseBody = DelLiveNoticeWidgetResponseBody;
class DelLiveNoticeWidgetResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DelLiveNoticeWidgetResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveNoticeWidgetResponse = DelLiveNoticeWidgetResponse;
class DeleteLiveHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveHeaders = DeleteLiveHeaders;
class DeleteLiveRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveRequest = DeleteLiveRequest;
class DeleteLiveResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: DeleteLiveResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveResponseBody = DeleteLiveResponseBody;
class DeleteLiveResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteLiveResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveResponse = DeleteLiveResponse;
class DeleteLiveFeedHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveFeedHeaders = DeleteLiveFeedHeaders;
class DeleteLiveFeedRequest extends $tea.Model {
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveFeedRequest = DeleteLiveFeedRequest;
class DeleteLiveFeedResponseBody extends $tea.Model {
    static names() {
        return {
            hasDelete: 'hasDelete',
        };
    }
    static types() {
        return {
            hasDelete: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveFeedResponseBody = DeleteLiveFeedResponseBody;
class DeleteLiveFeedResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeleteLiveFeedResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveFeedResponse = DeleteLiveFeedResponse;
class EditFeedReplayHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EditFeedReplayHeaders = EditFeedReplayHeaders;
class EditFeedReplayRequest extends $tea.Model {
    static names() {
        return {
            editEndTime: 'editEndTime',
            editStartTime: 'editStartTime',
            userId: 'userId',
        };
    }
    static types() {
        return {
            editEndTime: 'number',
            editStartTime: 'number',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EditFeedReplayRequest = EditFeedReplayRequest;
class EditFeedReplayResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EditFeedReplayResponseBody = EditFeedReplayResponseBody;
class EditFeedReplayResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: EditFeedReplayResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.EditFeedReplayResponse = EditFeedReplayResponse;
class GetGroupLiveListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetGroupLiveListHeaders = GetGroupLiveListHeaders;
class GetGroupLiveListRequest extends $tea.Model {
    static names() {
        return {
            requestBody: 'requestBody',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            requestBody: GetGroupLiveListRequestRequestBody,
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetGroupLiveListRequest = GetGroupLiveListRequest;
class GetGroupLiveListShrinkRequest extends $tea.Model {
    static names() {
        return {
            requestBodyShrink: 'requestBody',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            requestBodyShrink: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetGroupLiveListShrinkRequest = GetGroupLiveListShrinkRequest;
class GetGroupLiveListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetGroupLiveListResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetGroupLiveListResponseBody = GetGroupLiveListResponseBody;
class GetGroupLiveListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetGroupLiveListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetGroupLiveListResponse = GetGroupLiveListResponse;
class GetLiveReplayUrlHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetLiveReplayUrlHeaders = GetLiveReplayUrlHeaders;
class GetLiveReplayUrlRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetLiveReplayUrlRequest = GetLiveReplayUrlRequest;
class GetLiveReplayUrlResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetLiveReplayUrlResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetLiveReplayUrlResponseBody = GetLiveReplayUrlResponseBody;
class GetLiveReplayUrlResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetLiveReplayUrlResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetLiveReplayUrlResponse = GetLiveReplayUrlResponse;
class GetOrgLiveListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListHeaders = GetOrgLiveListHeaders;
class GetOrgLiveListRequest extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            requestBody: 'requestBody',
        };
    }
    static types() {
        return {
            corpId: 'string',
            requestBody: GetOrgLiveListRequestRequestBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListRequest = GetOrgLiveListRequest;
class GetOrgLiveListShrinkRequest extends $tea.Model {
    static names() {
        return {
            corpId: 'corpId',
            requestBodyShrink: 'requestBody',
        };
    }
    static types() {
        return {
            corpId: 'string',
            requestBodyShrink: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListShrinkRequest = GetOrgLiveListShrinkRequest;
class GetOrgLiveListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetOrgLiveListResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListResponseBody = GetOrgLiveListResponseBody;
class GetOrgLiveListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetOrgLiveListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListResponse = GetOrgLiveListResponse;
class GetUserAllLiveListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAllLiveListHeaders = GetUserAllLiveListHeaders;
class GetUserAllLiveListRequest extends $tea.Model {
    static names() {
        return {
            endTime: 'endTime',
            startTime: 'startTime',
            statuses: 'statuses',
            title: 'title',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            endTime: 'number',
            startTime: 'number',
            statuses: { 'type': 'array', 'itemType': 'number' },
            title: 'string',
            pageNumber: 'number',
            pageSize: 'number',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAllLiveListRequest = GetUserAllLiveListRequest;
class GetUserAllLiveListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetUserAllLiveListResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAllLiveListResponseBody = GetUserAllLiveListResponseBody;
class GetUserAllLiveListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetUserAllLiveListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAllLiveListResponse = GetUserAllLiveListResponse;
class GetUserCreateLiveListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserCreateLiveListHeaders = GetUserCreateLiveListHeaders;
class GetUserCreateLiveListRequest extends $tea.Model {
    static names() {
        return {
            endTime: 'endTime',
            startTime: 'startTime',
            statuses: 'statuses',
            title: 'title',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            endTime: 'number',
            startTime: 'number',
            statuses: { 'type': 'array', 'itemType': 'number' },
            title: 'string',
            maxResults: 'number',
            nextToken: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserCreateLiveListRequest = GetUserCreateLiveListRequest;
class GetUserCreateLiveListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetUserCreateLiveListResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserCreateLiveListResponseBody = GetUserCreateLiveListResponseBody;
class GetUserCreateLiveListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetUserCreateLiveListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserCreateLiveListResponse = GetUserCreateLiveListResponse;
class GetUserWatchLiveListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserWatchLiveListHeaders = GetUserWatchLiveListHeaders;
class GetUserWatchLiveListRequest extends $tea.Model {
    static names() {
        return {
            filterType: 'filterType',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            filterType: 'number',
            maxResults: 'number',
            nextToken: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserWatchLiveListRequest = GetUserWatchLiveListRequest;
class GetUserWatchLiveListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetUserWatchLiveListResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserWatchLiveListResponseBody = GetUserWatchLiveListResponseBody;
class GetUserWatchLiveListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetUserWatchLiveListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserWatchLiveListResponse = GetUserWatchLiveListResponse;
class ModifyFeedWhiteListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyFeedWhiteListHeaders = ModifyFeedWhiteListHeaders;
class ModifyFeedWhiteListRequest extends $tea.Model {
    static names() {
        return {
            action: 'action',
            modifyUserList: 'modifyUserList',
            userId: 'userId',
        };
    }
    static types() {
        return {
            action: 'number',
            modifyUserList: { 'type': 'array', 'itemType': 'string' },
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyFeedWhiteListRequest = ModifyFeedWhiteListRequest;
class ModifyFeedWhiteListShrinkRequest extends $tea.Model {
    static names() {
        return {
            action: 'action',
            modifyUserListShrink: 'modifyUserList',
            userId: 'userId',
        };
    }
    static types() {
        return {
            action: 'number',
            modifyUserListShrink: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyFeedWhiteListShrinkRequest = ModifyFeedWhiteListShrinkRequest;
class ModifyFeedWhiteListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyFeedWhiteListResponseBody = ModifyFeedWhiteListResponseBody;
class ModifyFeedWhiteListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ModifyFeedWhiteListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.ModifyFeedWhiteListResponse = ModifyFeedWhiteListResponse;
class QueryFeedWhiteListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryFeedWhiteListHeaders = QueryFeedWhiteListHeaders;
class QueryFeedWhiteListRequest extends $tea.Model {
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryFeedWhiteListRequest = QueryFeedWhiteListRequest;
class QueryFeedWhiteListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryFeedWhiteListResponseBody = QueryFeedWhiteListResponseBody;
class QueryFeedWhiteListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryFeedWhiteListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryFeedWhiteListResponse = QueryFeedWhiteListResponse;
class QueryLiveInfoHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInfoHeaders = QueryLiveInfoHeaders;
class QueryLiveInfoRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInfoRequest = QueryLiveInfoRequest;
class QueryLiveInfoResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryLiveInfoResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInfoResponseBody = QueryLiveInfoResponseBody;
class QueryLiveInfoResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryLiveInfoResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInfoResponse = QueryLiveInfoResponse;
class QueryLiveInteractionPluginHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInteractionPluginHeaders = QueryLiveInteractionPluginHeaders;
class QueryLiveInteractionPluginRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            pluginId: 'pluginId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            pluginId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInteractionPluginRequest = QueryLiveInteractionPluginRequest;
class QueryLiveInteractionPluginResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryLiveInteractionPluginResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInteractionPluginResponseBody = QueryLiveInteractionPluginResponseBody;
class QueryLiveInteractionPluginResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryLiveInteractionPluginResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInteractionPluginResponse = QueryLiveInteractionPluginResponse;
class QueryLiveWatchDetailHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchDetailHeaders = QueryLiveWatchDetailHeaders;
class QueryLiveWatchDetailRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchDetailRequest = QueryLiveWatchDetailRequest;
class QueryLiveWatchDetailResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryLiveWatchDetailResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchDetailResponseBody = QueryLiveWatchDetailResponseBody;
class QueryLiveWatchDetailResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryLiveWatchDetailResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchDetailResponse = QueryLiveWatchDetailResponse;
class QueryLiveWatchUserListHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchUserListHeaders = QueryLiveWatchUserListHeaders;
class QueryLiveWatchUserListRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            pageNumber: 'number',
            pageSize: 'number',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchUserListRequest = QueryLiveWatchUserListRequest;
class QueryLiveWatchUserListResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryLiveWatchUserListResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchUserListResponseBody = QueryLiveWatchUserListResponseBody;
class QueryLiveWatchUserListResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryLiveWatchUserListResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchUserListResponse = QueryLiveWatchUserListResponse;
class QuerySubscribeStatusHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QuerySubscribeStatusHeaders = QuerySubscribeStatusHeaders;
class QuerySubscribeStatusRequest extends $tea.Model {
    static names() {
        return {
            body: 'body',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            body: QuerySubscribeStatusRequestBody,
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QuerySubscribeStatusRequest = QuerySubscribeStatusRequest;
class QuerySubscribeStatusShrinkRequest extends $tea.Model {
    static names() {
        return {
            bodyShrink: 'body',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            bodyShrink: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QuerySubscribeStatusShrinkRequest = QuerySubscribeStatusShrinkRequest;
class QuerySubscribeStatusResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QuerySubscribeStatusResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QuerySubscribeStatusResponseBody = QuerySubscribeStatusResponseBody;
class QuerySubscribeStatusResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QuerySubscribeStatusResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QuerySubscribeStatusResponse = QuerySubscribeStatusResponse;
class SendLiveInteractionPluginEffectsMsgHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveInteractionPluginEffectsMsgHeaders = SendLiveInteractionPluginEffectsMsgHeaders;
class SendLiveInteractionPluginEffectsMsgRequest extends $tea.Model {
    static names() {
        return {
            count: 'count',
            lottieFileUrl: 'lottieFileUrl',
            msgIconUrl: 'msgIconUrl',
            msgText: 'msgText',
            pluginSubId: 'pluginSubId',
            senderUnionId: 'senderUnionId',
            liveId: 'liveId',
            pluginId: 'pluginId',
        };
    }
    static types() {
        return {
            count: 'number',
            lottieFileUrl: 'string',
            msgIconUrl: 'string',
            msgText: 'string',
            pluginSubId: 'string',
            senderUnionId: 'string',
            liveId: 'string',
            pluginId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveInteractionPluginEffectsMsgRequest = SendLiveInteractionPluginEffectsMsgRequest;
class SendLiveInteractionPluginEffectsMsgResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: SendLiveInteractionPluginEffectsMsgResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveInteractionPluginEffectsMsgResponseBody = SendLiveInteractionPluginEffectsMsgResponseBody;
class SendLiveInteractionPluginEffectsMsgResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendLiveInteractionPluginEffectsMsgResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveInteractionPluginEffectsMsgResponse = SendLiveInteractionPluginEffectsMsgResponse;
class SendLivePluginUserActionMsgHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLivePluginUserActionMsgHeaders = SendLivePluginUserActionMsgHeaders;
class SendLivePluginUserActionMsgRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            pluginEffectsMessage: 'pluginEffectsMessage',
            pluginId: 'pluginId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            pluginEffectsMessage: SendLivePluginUserActionMsgRequestPluginEffectsMessage,
            pluginId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLivePluginUserActionMsgRequest = SendLivePluginUserActionMsgRequest;
class SendLivePluginUserActionMsgShrinkRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            pluginEffectsMessageShrink: 'pluginEffectsMessage',
            pluginId: 'pluginId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            pluginEffectsMessageShrink: 'string',
            pluginId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLivePluginUserActionMsgShrinkRequest = SendLivePluginUserActionMsgShrinkRequest;
class SendLivePluginUserActionMsgResponseBody extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLivePluginUserActionMsgResponseBody = SendLivePluginUserActionMsgResponseBody;
class SendLivePluginUserActionMsgResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendLivePluginUserActionMsgResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLivePluginUserActionMsgResponse = SendLivePluginUserActionMsgResponse;
class StartCloudFeedHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.StartCloudFeedHeaders = StartCloudFeedHeaders;
class StartCloudFeedRequest extends $tea.Model {
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.StartCloudFeedRequest = StartCloudFeedRequest;
class StartCloudFeedResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.StartCloudFeedResponseBody = StartCloudFeedResponseBody;
class StartCloudFeedResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: StartCloudFeedResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.StartCloudFeedResponse = StartCloudFeedResponse;
class StopCloudFeedHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.StopCloudFeedHeaders = StopCloudFeedHeaders;
class StopCloudFeedRequest extends $tea.Model {
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.StopCloudFeedRequest = StopCloudFeedRequest;
class StopCloudFeedResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.StopCloudFeedResponseBody = StopCloudFeedResponseBody;
class StopCloudFeedResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: StopCloudFeedResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.StopCloudFeedResponse = StopCloudFeedResponse;
class SubscribeLiveHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SubscribeLiveHeaders = SubscribeLiveHeaders;
class SubscribeLiveRequest extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            subscribe: 'subscribe',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            liveId: 'string',
            subscribe: 'boolean',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SubscribeLiveRequest = SubscribeLiveRequest;
class SubscribeLiveResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: SubscribeLiveResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SubscribeLiveResponseBody = SubscribeLiveResponseBody;
class SubscribeLiveResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SubscribeLiveResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SubscribeLiveResponse = SubscribeLiveResponse;
class UpdateLiveHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveHeaders = UpdateLiveHeaders;
class UpdateLiveRequest extends $tea.Model {
    static names() {
        return {
            coverUrl: 'coverUrl',
            introduction: 'introduction',
            liveId: 'liveId',
            preEndTime: 'preEndTime',
            preStartTime: 'preStartTime',
            title: 'title',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            coverUrl: 'string',
            introduction: 'string',
            liveId: 'string',
            preEndTime: 'number',
            preStartTime: 'number',
            title: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveRequest = UpdateLiveRequest;
class UpdateLiveResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: UpdateLiveResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveResponseBody = UpdateLiveResponseBody;
class UpdateLiveResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateLiveResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveResponse = UpdateLiveResponse;
class UpdateLiveFeedHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveFeedHeaders = UpdateLiveFeedHeaders;
class UpdateLiveFeedRequest extends $tea.Model {
    static names() {
        return {
            coverUrl: 'coverUrl',
            introduction: 'introduction',
            startTime: 'startTime',
            title: 'title',
            userId: 'userId',
        };
    }
    static types() {
        return {
            coverUrl: 'string',
            introduction: 'string',
            startTime: 'number',
            title: 'string',
            userId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveFeedRequest = UpdateLiveFeedRequest;
class UpdateLiveFeedResponseBody extends $tea.Model {
    static names() {
        return {
            hasUpdate: 'hasUpdate',
        };
    }
    static types() {
        return {
            hasUpdate: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveFeedResponseBody = UpdateLiveFeedResponseBody;
class UpdateLiveFeedResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateLiveFeedResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveFeedResponse = UpdateLiveFeedResponse;
class UpdateLiveInteractionPluginHeaders extends $tea.Model {
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveInteractionPluginHeaders = UpdateLiveInteractionPluginHeaders;
class UpdateLiveInteractionPluginRequest extends $tea.Model {
    static names() {
        return {
            anchorJumpUrl: 'anchorJumpUrl',
            pluginIconUrl: 'pluginIconUrl',
            pluginName: 'pluginName',
            pluginNameEn: 'pluginNameEn',
            viewerJumpUrl: 'viewerJumpUrl',
            liveId: 'liveId',
            pluginId: 'pluginId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            anchorJumpUrl: 'string',
            pluginIconUrl: 'string',
            pluginName: 'string',
            pluginNameEn: 'string',
            viewerJumpUrl: 'string',
            liveId: 'string',
            pluginId: 'string',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveInteractionPluginRequest = UpdateLiveInteractionPluginRequest;
class UpdateLiveInteractionPluginResponseBody extends $tea.Model {
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: UpdateLiveInteractionPluginResponseBodyResult,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveInteractionPluginResponseBody = UpdateLiveInteractionPluginResponseBody;
class UpdateLiveInteractionPluginResponse extends $tea.Model {
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateLiveInteractionPluginResponseBody,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveInteractionPluginResponse = UpdateLiveInteractionPluginResponse;
class AddLiveInteractionPluginResponseBodyResult extends $tea.Model {
    static names() {
        return {
            pluginId: 'pluginId',
        };
    }
    static types() {
        return {
            pluginId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveInteractionPluginResponseBodyResult = AddLiveInteractionPluginResponseBodyResult;
class AddLiveNoticeWidgetResponseBodyResult extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.AddLiveNoticeWidgetResponseBodyResult = AddLiveNoticeWidgetResponseBodyResult;
class CreateLiveResponseBodyResult extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
        };
    }
    static types() {
        return {
            liveId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.CreateLiveResponseBodyResult = CreateLiveResponseBodyResult;
class DelLiveInteractionPluginResponseBodyResult extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveInteractionPluginResponseBodyResult = DelLiveInteractionPluginResponseBodyResult;
class DelLiveNoticeWidgetResponseBodyResult extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DelLiveNoticeWidgetResponseBodyResult = DelLiveNoticeWidgetResponseBodyResult;
class DeleteLiveResponseBodyResult extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.DeleteLiveResponseBodyResult = DeleteLiveResponseBodyResult;
class GetGroupLiveListRequestRequestBody extends $tea.Model {
    static names() {
        return {
            endTime: 'endTime',
            openConversationId: 'openConversationId',
            startTime: 'startTime',
        };
    }
    static types() {
        return {
            endTime: 'number',
            openConversationId: 'string',
            startTime: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetGroupLiveListRequestRequestBody = GetGroupLiveListRequestRequestBody;
class GetGroupLiveListResponseBodyResultGroupLiveList extends $tea.Model {
    static names() {
        return {
            anchorNickname: 'anchorNickname',
            anchorUnionId: 'anchorUnionId',
            liveEndTime: 'liveEndTime',
            liveStartTime: 'liveStartTime',
            liveUuid: 'liveUuid',
            title: 'title',
        };
    }
    static types() {
        return {
            anchorNickname: 'string',
            anchorUnionId: 'string',
            liveEndTime: 'number',
            liveStartTime: 'number',
            liveUuid: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetGroupLiveListResponseBodyResultGroupLiveList = GetGroupLiveListResponseBodyResultGroupLiveList;
class GetGroupLiveListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            groupLiveList: 'groupLiveList',
        };
    }
    static types() {
        return {
            groupLiveList: { 'type': 'array', 'itemType': GetGroupLiveListResponseBodyResultGroupLiveList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetGroupLiveListResponseBodyResult = GetGroupLiveListResponseBodyResult;
class GetLiveReplayUrlResponseBodyResult extends $tea.Model {
    static names() {
        return {
            replayUrl: 'replayUrl',
        };
    }
    static types() {
        return {
            replayUrl: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetLiveReplayUrlResponseBodyResult = GetLiveReplayUrlResponseBodyResult;
class GetOrgLiveListRequestRequestBody extends $tea.Model {
    static names() {
        return {
            endTime: 'endTime',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            startTime: 'startTime',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            endTime: 'number',
            pageNumber: 'number',
            pageSize: 'number',
            startTime: 'number',
            unionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListRequestRequestBody = GetOrgLiveListRequestRequestBody;
class GetOrgLiveListResponseBodyResultNewLiveLiveList extends $tea.Model {
    static names() {
        return {
            anchorNickname: 'anchorNickname',
            anchorUnionId: 'anchorUnionId',
            liveEndTime: 'liveEndTime',
            liveStartTime: 'liveStartTime',
            liveUuid: 'liveUuid',
            shareOpenConversationIds: 'shareOpenConversationIds',
            title: 'title',
        };
    }
    static types() {
        return {
            anchorNickname: 'string',
            anchorUnionId: 'string',
            liveEndTime: 'number',
            liveStartTime: 'number',
            liveUuid: 'string',
            shareOpenConversationIds: { 'type': 'array', 'itemType': 'string' },
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListResponseBodyResultNewLiveLiveList = GetOrgLiveListResponseBodyResultNewLiveLiveList;
class GetOrgLiveListResponseBodyResultNewLive extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            liveList: 'liveList',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            liveList: { 'type': 'array', 'itemType': GetOrgLiveListResponseBodyResultNewLiveLiveList },
            pageNumber: 'number',
            pageSize: 'number',
            totalCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListResponseBodyResultNewLive = GetOrgLiveListResponseBodyResultNewLive;
class GetOrgLiveListResponseBodyResultUpdateLiveLiveList extends $tea.Model {
    static names() {
        return {
            anchorNickname: 'anchorNickname',
            anchorUnionId: 'anchorUnionId',
            liveEndTime: 'liveEndTime',
            liveStartTime: 'liveStartTime',
            liveUuid: 'liveUuid',
            title: 'title',
        };
    }
    static types() {
        return {
            anchorNickname: 'string',
            anchorUnionId: 'string',
            liveEndTime: 'number',
            liveStartTime: 'number',
            liveUuid: 'string',
            title: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListResponseBodyResultUpdateLiveLiveList = GetOrgLiveListResponseBodyResultUpdateLiveLiveList;
class GetOrgLiveListResponseBodyResultUpdateLive extends $tea.Model {
    static names() {
        return {
            hasMore: 'hasMore',
            liveList: 'liveList',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            liveList: { 'type': 'array', 'itemType': GetOrgLiveListResponseBodyResultUpdateLiveLiveList },
            pageNumber: 'number',
            pageSize: 'number',
            totalCount: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListResponseBodyResultUpdateLive = GetOrgLiveListResponseBodyResultUpdateLive;
class GetOrgLiveListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            newLive: 'newLive',
            updateLive: 'updateLive',
        };
    }
    static types() {
        return {
            newLive: GetOrgLiveListResponseBodyResultNewLive,
            updateLive: GetOrgLiveListResponseBodyResultUpdateLive,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetOrgLiveListResponseBodyResult = GetOrgLiveListResponseBodyResult;
class GetUserAllLiveListResponseBodyResultLiveInfoPopModelListExtraInfo extends $tea.Model {
    static names() {
        return {
            hasSubscribed: 'hasSubscribed',
            isForecastExpired: 'isForecastExpired',
            watchProgressMs: 'watchProgressMs',
        };
    }
    static types() {
        return {
            hasSubscribed: 'boolean',
            isForecastExpired: 'boolean',
            watchProgressMs: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAllLiveListResponseBodyResultLiveInfoPopModelListExtraInfo = GetUserAllLiveListResponseBodyResultLiveInfoPopModelListExtraInfo;
class GetUserAllLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo extends $tea.Model {
    static names() {
        return {
            coverUrl: 'coverUrl',
            duration: 'duration',
            endTime: 'endTime',
            introduction: 'introduction',
            liveId: 'liveId',
            livePlayUrl: 'livePlayUrl',
            liveStatus: 'liveStatus',
            startTime: 'startTime',
            subscribeCount: 'subscribeCount',
            title: 'title',
            unionId: 'unionId',
            uv: 'uv',
        };
    }
    static types() {
        return {
            coverUrl: 'string',
            duration: 'number',
            endTime: 'number',
            introduction: 'string',
            liveId: 'string',
            livePlayUrl: 'string',
            liveStatus: 'number',
            startTime: 'number',
            subscribeCount: 'number',
            title: 'string',
            unionId: 'string',
            uv: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAllLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo = GetUserAllLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo;
class GetUserAllLiveListResponseBodyResultLiveInfoPopModelList extends $tea.Model {
    static names() {
        return {
            extraInfo: 'extraInfo',
            liveBasicInfo: 'liveBasicInfo',
        };
    }
    static types() {
        return {
            extraInfo: GetUserAllLiveListResponseBodyResultLiveInfoPopModelListExtraInfo,
            liveBasicInfo: GetUserAllLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAllLiveListResponseBodyResultLiveInfoPopModelList = GetUserAllLiveListResponseBodyResultLiveInfoPopModelList;
class GetUserAllLiveListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            hasFinish: 'hasFinish',
            liveInfoPopModelList: 'liveInfoPopModelList',
        };
    }
    static types() {
        return {
            hasFinish: 'boolean',
            liveInfoPopModelList: { 'type': 'array', 'itemType': GetUserAllLiveListResponseBodyResultLiveInfoPopModelList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserAllLiveListResponseBodyResult = GetUserAllLiveListResponseBodyResult;
class GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListHasSubscribed extends $tea.Model {
    static names() {
        return {
            hasSubscribed: 'hasSubscribed',
            isForecastExpired: 'isForecastExpired',
            watchProgressMs: 'watchProgressMs',
        };
    }
    static types() {
        return {
            hasSubscribed: 'boolean',
            isForecastExpired: 'boolean',
            watchProgressMs: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListHasSubscribed = GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListHasSubscribed;
class GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo extends $tea.Model {
    static names() {
        return {
            coverUrl: 'coverUrl',
            duration: 'duration',
            endTime: 'endTime',
            introduction: 'introduction',
            liveId: 'liveId',
            livePlayUrl: 'livePlayUrl',
            liveStatus: 'liveStatus',
            startTime: 'startTime',
            subscribeCount: 'subscribeCount',
            title: 'title',
            unionId: 'unionId',
            uv: 'uv',
        };
    }
    static types() {
        return {
            coverUrl: 'string',
            duration: 'number',
            endTime: 'number',
            introduction: 'string',
            liveId: 'string',
            livePlayUrl: 'string',
            liveStatus: 'number',
            startTime: 'number',
            subscribeCount: 'number',
            title: 'string',
            unionId: 'string',
            uv: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo = GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo;
class GetUserCreateLiveListResponseBodyResultLiveInfoPopModelList extends $tea.Model {
    static names() {
        return {
            hasSubscribed: 'hasSubscribed',
            liveBasicInfo: 'liveBasicInfo',
        };
    }
    static types() {
        return {
            hasSubscribed: GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListHasSubscribed,
            liveBasicInfo: GetUserCreateLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserCreateLiveListResponseBodyResultLiveInfoPopModelList = GetUserCreateLiveListResponseBodyResultLiveInfoPopModelList;
class GetUserCreateLiveListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            hasFinish: 'hasFinish',
            liveInfoPopModelList: 'liveInfoPopModelList',
            nextToken: 'nextToken',
            total: 'total',
        };
    }
    static types() {
        return {
            hasFinish: 'boolean',
            liveInfoPopModelList: { 'type': 'array', 'itemType': GetUserCreateLiveListResponseBodyResultLiveInfoPopModelList },
            nextToken: 'string',
            total: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserCreateLiveListResponseBodyResult = GetUserCreateLiveListResponseBodyResult;
class GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListExtraInfo extends $tea.Model {
    static names() {
        return {
            hasSubscribed: 'hasSubscribed',
            isForecastExpired: 'isForecastExpired',
            watchProgressMs: 'watchProgressMs',
        };
    }
    static types() {
        return {
            hasSubscribed: 'boolean',
            isForecastExpired: 'boolean',
            watchProgressMs: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListExtraInfo = GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListExtraInfo;
class GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo extends $tea.Model {
    static names() {
        return {
            coverUrl: 'coverUrl',
            duration: 'duration',
            endTime: 'endTime',
            introduction: 'introduction',
            liveId: 'liveId',
            livePlayUrl: 'livePlayUrl',
            liveStatus: 'liveStatus',
            startTime: 'startTime',
            subscribeCount: 'subscribeCount',
            title: 'title',
            unionId: 'unionId',
            uv: 'uv',
        };
    }
    static types() {
        return {
            coverUrl: 'string',
            duration: 'number',
            endTime: 'number',
            introduction: 'string',
            liveId: 'string',
            livePlayUrl: 'string',
            liveStatus: 'number',
            startTime: 'number',
            subscribeCount: 'number',
            title: 'string',
            unionId: 'string',
            uv: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo = GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo;
class GetUserWatchLiveListResponseBodyResultLiveInfoPopModelList extends $tea.Model {
    static names() {
        return {
            extraInfo: 'extraInfo',
            liveBasicInfo: 'liveBasicInfo',
        };
    }
    static types() {
        return {
            extraInfo: GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListExtraInfo,
            liveBasicInfo: GetUserWatchLiveListResponseBodyResultLiveInfoPopModelListLiveBasicInfo,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserWatchLiveListResponseBodyResultLiveInfoPopModelList = GetUserWatchLiveListResponseBodyResultLiveInfoPopModelList;
class GetUserWatchLiveListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            hasFinish: 'hasFinish',
            liveInfoPopModelList: 'liveInfoPopModelList',
            nextToken: 'nextToken',
            total: 'total',
        };
    }
    static types() {
        return {
            hasFinish: 'boolean',
            liveInfoPopModelList: { 'type': 'array', 'itemType': GetUserWatchLiveListResponseBodyResultLiveInfoPopModelList },
            nextToken: 'string',
            total: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUserWatchLiveListResponseBodyResult = GetUserWatchLiveListResponseBodyResult;
class QueryLiveInfoResponseBodyResultLiveInfo extends $tea.Model {
    static names() {
        return {
            coverUrl: 'coverUrl',
            duration: 'duration',
            endTime: 'endTime',
            introduction: 'introduction',
            liveId: 'liveId',
            livePlayUrl: 'livePlayUrl',
            liveStatus: 'liveStatus',
            playbackDuration: 'playbackDuration',
            startTime: 'startTime',
            subscribeCount: 'subscribeCount',
            title: 'title',
            unionId: 'unionId',
            uv: 'uv',
        };
    }
    static types() {
        return {
            coverUrl: 'string',
            duration: 'number',
            endTime: 'number',
            introduction: 'string',
            liveId: 'string',
            livePlayUrl: 'string',
            liveStatus: 'number',
            playbackDuration: 'number',
            startTime: 'number',
            subscribeCount: 'number',
            title: 'string',
            unionId: 'string',
            uv: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInfoResponseBodyResultLiveInfo = QueryLiveInfoResponseBodyResultLiveInfo;
class QueryLiveInfoResponseBodyResult extends $tea.Model {
    static names() {
        return {
            liveInfo: 'liveInfo',
        };
    }
    static types() {
        return {
            liveInfo: QueryLiveInfoResponseBodyResultLiveInfo,
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInfoResponseBodyResult = QueryLiveInfoResponseBodyResult;
class QueryLiveInteractionPluginResponseBodyResultLivePluginList extends $tea.Model {
    static names() {
        return {
            anchorJumpUrl: 'anchorJumpUrl',
            pluginIconUrl: 'pluginIconUrl',
            pluginId: 'pluginId',
            pluginName: 'pluginName',
            pluginNameEn: 'pluginNameEn',
            viewerJumpUrl: 'viewerJumpUrl',
        };
    }
    static types() {
        return {
            anchorJumpUrl: 'string',
            pluginIconUrl: 'string',
            pluginId: 'string',
            pluginName: 'string',
            pluginNameEn: 'string',
            viewerJumpUrl: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInteractionPluginResponseBodyResultLivePluginList = QueryLiveInteractionPluginResponseBodyResultLivePluginList;
class QueryLiveInteractionPluginResponseBodyResult extends $tea.Model {
    static names() {
        return {
            livePluginList: 'livePluginList',
        };
    }
    static types() {
        return {
            livePluginList: { 'type': 'array', 'itemType': QueryLiveInteractionPluginResponseBodyResultLivePluginList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveInteractionPluginResponseBodyResult = QueryLiveInteractionPluginResponseBodyResult;
class QueryLiveWatchDetailResponseBodyResult extends $tea.Model {
    static names() {
        return {
            avgWatchTime: 'avgWatchTime',
            liveUv: 'liveUv',
            msgCount: 'msgCount',
            playbackUv: 'playbackUv',
            praiseCount: 'praiseCount',
            pv: 'pv',
            totalWatchTime: 'totalWatchTime',
            uv: 'uv',
        };
    }
    static types() {
        return {
            avgWatchTime: 'number',
            liveUv: 'number',
            msgCount: 'number',
            playbackUv: 'number',
            praiseCount: 'number',
            pv: 'number',
            totalWatchTime: 'number',
            uv: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchDetailResponseBodyResult = QueryLiveWatchDetailResponseBodyResult;
class QueryLiveWatchUserListResponseBodyResultOrgUsesList extends $tea.Model {
    static names() {
        return {
            deptName: 'deptName',
            name: 'name',
            unionId: 'unionId',
            userId: 'userId',
            watchLiveTime: 'watchLiveTime',
            watchPlaybackTime: 'watchPlaybackTime',
            watchProgressMs: 'watchProgressMs',
        };
    }
    static types() {
        return {
            deptName: 'string',
            name: 'string',
            unionId: 'string',
            userId: 'string',
            watchLiveTime: 'number',
            watchPlaybackTime: 'number',
            watchProgressMs: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchUserListResponseBodyResultOrgUsesList = QueryLiveWatchUserListResponseBodyResultOrgUsesList;
class QueryLiveWatchUserListResponseBodyResultOutOrgUserList extends $tea.Model {
    static names() {
        return {
            name: 'name',
            watchLiveTime: 'watchLiveTime',
            watchPlaybackTime: 'watchPlaybackTime',
            watchProgressMs: 'watchProgressMs',
        };
    }
    static types() {
        return {
            name: 'string',
            watchLiveTime: 'number',
            watchPlaybackTime: 'number',
            watchProgressMs: 'number',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchUserListResponseBodyResultOutOrgUserList = QueryLiveWatchUserListResponseBodyResultOutOrgUserList;
class QueryLiveWatchUserListResponseBodyResult extends $tea.Model {
    static names() {
        return {
            orgUsesList: 'orgUsesList',
            outOrgUserList: 'outOrgUserList',
        };
    }
    static types() {
        return {
            orgUsesList: { 'type': 'array', 'itemType': QueryLiveWatchUserListResponseBodyResultOrgUsesList },
            outOrgUserList: { 'type': 'array', 'itemType': QueryLiveWatchUserListResponseBodyResultOutOrgUserList },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryLiveWatchUserListResponseBodyResult = QueryLiveWatchUserListResponseBodyResult;
class QuerySubscribeStatusRequestBody extends $tea.Model {
    static names() {
        return {
            liveIds: 'liveIds',
        };
    }
    static types() {
        return {
            liveIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QuerySubscribeStatusRequestBody = QuerySubscribeStatusRequestBody;
class QuerySubscribeStatusResponseBodyResultSubscribeStatusDTOS extends $tea.Model {
    static names() {
        return {
            liveId: 'liveId',
            subscribe: 'subscribe',
        };
    }
    static types() {
        return {
            liveId: 'string',
            subscribe: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QuerySubscribeStatusResponseBodyResultSubscribeStatusDTOS = QuerySubscribeStatusResponseBodyResultSubscribeStatusDTOS;
class QuerySubscribeStatusResponseBodyResult extends $tea.Model {
    static names() {
        return {
            subscribeStatusDTOS: 'subscribeStatusDTOS',
        };
    }
    static types() {
        return {
            subscribeStatusDTOS: { 'type': 'array', 'itemType': QuerySubscribeStatusResponseBodyResultSubscribeStatusDTOS },
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.QuerySubscribeStatusResponseBodyResult = QuerySubscribeStatusResponseBodyResult;
class SendLiveInteractionPluginEffectsMsgResponseBodyResult extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLiveInteractionPluginEffectsMsgResponseBodyResult = SendLiveInteractionPluginEffectsMsgResponseBodyResult;
class SendLivePluginUserActionMsgRequestPluginEffectsMessage extends $tea.Model {
    static names() {
        return {
            count: 'count',
            lottieFileUrl: 'lottieFileUrl',
            msgIconUrl: 'msgIconUrl',
            msgText: 'msgText',
            pluginSubId: 'pluginSubId',
            senderUnionId: 'senderUnionId',
        };
    }
    static types() {
        return {
            count: 'number',
            lottieFileUrl: 'string',
            msgIconUrl: 'string',
            msgText: 'string',
            pluginSubId: 'string',
            senderUnionId: 'string',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SendLivePluginUserActionMsgRequestPluginEffectsMessage = SendLivePluginUserActionMsgRequestPluginEffectsMessage;
class SubscribeLiveResponseBodyResult extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.SubscribeLiveResponseBodyResult = SubscribeLiveResponseBodyResult;
class UpdateLiveResponseBodyResult extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveResponseBodyResult = UpdateLiveResponseBodyResult;
class UpdateLiveInteractionPluginResponseBodyResult extends $tea.Model {
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateLiveInteractionPluginResponseBodyResult = UpdateLiveInteractionPluginResponseBodyResult;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        let gatewayClient = new gateway_dingtalk_1.default();
        this._spi = gatewayClient;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    /**
     * 
     *
     * @param request - AddLiveInteractionPluginRequest
     * @param headers - AddLiveInteractionPluginHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns AddLiveInteractionPluginResponse
     */
    async addLiveInteractionPluginWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.anchorJumpUrl)) {
            body["anchorJumpUrl"] = request.anchorJumpUrl;
        }
        if (!tea_util_1.default.isUnset(request.pluginIconUrl)) {
            body["pluginIconUrl"] = request.pluginIconUrl;
        }
        if (!tea_util_1.default.isUnset(request.pluginName)) {
            body["pluginName"] = request.pluginName;
        }
        if (!tea_util_1.default.isUnset(request.pluginNameEn)) {
            body["pluginNameEn"] = request.pluginNameEn;
        }
        if (!tea_util_1.default.isUnset(request.viewerJumpUrl)) {
            body["viewerJumpUrl"] = request.viewerJumpUrl;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddLiveInteractionPlugin",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/interactionPlugins`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddLiveInteractionPluginResponse({}));
    }
    /**
     * 
     *
     * @param request - AddLiveInteractionPluginRequest
     * @returns AddLiveInteractionPluginResponse
     */
    async addLiveInteractionPlugin(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddLiveInteractionPluginHeaders({});
        return await this.addLiveInteractionPluginWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - AddLiveNoticeWidgetRequest
     * @param headers - AddLiveNoticeWidgetHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns AddLiveNoticeWidgetResponse
     */
    async addLiveNoticeWidgetWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.iconUrl)) {
            query["iconUrl"] = request.iconUrl;
        }
        if (!tea_util_1.default.isUnset(request.jumpUrl)) {
            query["jumpUrl"] = request.jumpUrl;
        }
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.noticeText)) {
            query["noticeText"] = request.noticeText;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "AddLiveNoticeWidget",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/noticeWidgets`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddLiveNoticeWidgetResponse({}));
    }
    /**
     * 
     *
     * @param request - AddLiveNoticeWidgetRequest
     * @returns AddLiveNoticeWidgetResponse
     */
    async addLiveNoticeWidget(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddLiveNoticeWidgetHeaders({});
        return await this.addLiveNoticeWidgetWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - AddShareCidListRequest
     * @param headers - AddShareCidListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns AddShareCidListResponse
     */
    async addShareCidListWithOptions(feedId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupIdType)) {
            body["groupIdType"] = request.groupIdType;
        }
        if (!tea_util_1.default.isUnset(request.groupIds)) {
            body["groupIds"] = request.groupIds;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddShareCidList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/cloudFeeds/${feedId}/share`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddShareCidListResponse({}));
    }
    /**
     * 
     *
     * @param request - AddShareCidListRequest
     * @returns AddShareCidListResponse
     */
    async addShareCidList(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddShareCidListHeaders({});
        return await this.addShareCidListWithOptions(feedId, request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - CreateCloudFeedRequest
     * @param headers - CreateCloudFeedHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns CreateCloudFeedResponse
     */
    async createCloudFeedWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.coverUrl)) {
            body["coverUrl"] = request.coverUrl;
        }
        if (!tea_util_1.default.isUnset(request.intro)) {
            body["intro"] = request.intro;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.videoUrl)) {
            body["videoUrl"] = request.videoUrl;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateCloudFeed",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/cloudFeeds`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateCloudFeedResponse({}));
    }
    /**
     * 
     *
     * @param request - CreateCloudFeedRequest
     * @returns CreateCloudFeedResponse
     */
    async createCloudFeed(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateCloudFeedHeaders({});
        return await this.createCloudFeedWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - CreateLiveRequest
     * @param headers - CreateLiveHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns CreateLiveResponse
     */
    async createLiveWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.coverUrl)) {
            body["coverUrl"] = request.coverUrl;
        }
        if (!tea_util_1.default.isUnset(request.introduction)) {
            body["introduction"] = request.introduction;
        }
        if (!tea_util_1.default.isUnset(request.preEndTime)) {
            body["preEndTime"] = request.preEndTime;
        }
        if (!tea_util_1.default.isUnset(request.preStartTime)) {
            body["preStartTime"] = request.preStartTime;
        }
        if (!tea_util_1.default.isUnset(request.publicType)) {
            body["publicType"] = request.publicType;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateLive",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/lives`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateLiveResponse({}));
    }
    /**
     * 
     *
     * @param request - CreateLiveRequest
     * @returns CreateLiveResponse
     */
    async createLive(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateLiveHeaders({});
        return await this.createLiveWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - DelLiveInteractionPluginRequest
     * @param headers - DelLiveInteractionPluginHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns DelLiveInteractionPluginResponse
     */
    async delLiveInteractionPluginWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.pluginId)) {
            query["pluginId"] = request.pluginId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DelLiveInteractionPlugin",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/interactionPlugins`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DelLiveInteractionPluginResponse({}));
    }
    /**
     * 
     *
     * @param request - DelLiveInteractionPluginRequest
     * @returns DelLiveInteractionPluginResponse
     */
    async delLiveInteractionPlugin(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DelLiveInteractionPluginHeaders({});
        return await this.delLiveInteractionPluginWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - DelLiveNoticeWidgetRequest
     * @param headers - DelLiveNoticeWidgetHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns DelLiveNoticeWidgetResponse
     */
    async delLiveNoticeWidgetWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DelLiveNoticeWidget",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/noticeWidgets`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DelLiveNoticeWidgetResponse({}));
    }
    /**
     * 
     *
     * @param request - DelLiveNoticeWidgetRequest
     * @returns DelLiveNoticeWidgetResponse
     */
    async delLiveNoticeWidget(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DelLiveNoticeWidgetHeaders({});
        return await this.delLiveNoticeWidgetWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - DeleteLiveRequest
     * @param headers - DeleteLiveHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns DeleteLiveResponse
     */
    async deleteLiveWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteLive",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/lives`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteLiveResponse({}));
    }
    /**
     * 
     *
     * @param request - DeleteLiveRequest
     * @returns DeleteLiveResponse
     */
    async deleteLive(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteLiveHeaders({});
        return await this.deleteLiveWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - DeleteLiveFeedRequest
     * @param headers - DeleteLiveFeedHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns DeleteLiveFeedResponse
     */
    async deleteLiveFeedWithOptions(feedId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "DeleteLiveFeed",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/openFeeds/${feedId}`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeleteLiveFeedResponse({}));
    }
    /**
     * 
     *
     * @param request - DeleteLiveFeedRequest
     * @returns DeleteLiveFeedResponse
     */
    async deleteLiveFeed(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteLiveFeedHeaders({});
        return await this.deleteLiveFeedWithOptions(feedId, request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - EditFeedReplayRequest
     * @param headers - EditFeedReplayHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns EditFeedReplayResponse
     */
    async editFeedReplayWithOptions(feedId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.editEndTime)) {
            body["editEndTime"] = request.editEndTime;
        }
        if (!tea_util_1.default.isUnset(request.editStartTime)) {
            body["editStartTime"] = request.editStartTime;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "EditFeedReplay",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/openFeeds/${feedId}/cutReplay`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new EditFeedReplayResponse({}));
    }
    /**
     * 
     *
     * @param request - EditFeedReplayRequest
     * @returns EditFeedReplayResponse
     */
    async editFeedReplay(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new EditFeedReplayHeaders({});
        return await this.editFeedReplayWithOptions(feedId, request, headers, runtime);
    }
    /**
     * 
     *
     * @param tmpReq - GetGroupLiveListRequest
     * @param headers - GetGroupLiveListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns GetGroupLiveListResponse
     */
    async getGroupLiveListWithOptions(tmpReq, headers, runtime) {
        tea_util_1.default.validateModel(tmpReq);
        let request = new GetGroupLiveListShrinkRequest({});
        openapi_util_1.default.convert(tmpReq, request);
        if (!tea_util_1.default.isUnset(tmpReq.requestBody)) {
            request.requestBodyShrink = openapi_util_1.default.arrayToStringWithSpecifiedStyle(tmpReq.requestBody, "requestBody", "json");
        }
        let query = {};
        if (!tea_util_1.default.isUnset(request.requestBodyShrink)) {
            query["requestBody"] = request.requestBodyShrink;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetGroupLiveList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/groupLives`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetGroupLiveListResponse({}));
    }
    /**
     * 
     *
     * @param request - GetGroupLiveListRequest
     * @returns GetGroupLiveListResponse
     */
    async getGroupLiveList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetGroupLiveListHeaders({});
        return await this.getGroupLiveListWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - GetLiveReplayUrlRequest
     * @param headers - GetLiveReplayUrlHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns GetLiveReplayUrlResponse
     */
    async getLiveReplayUrlWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetLiveReplayUrl",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/lives/replayUrls`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetLiveReplayUrlResponse({}));
    }
    /**
     * 
     *
     * @param request - GetLiveReplayUrlRequest
     * @returns GetLiveReplayUrlResponse
     */
    async getLiveReplayUrl(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetLiveReplayUrlHeaders({});
        return await this.getLiveReplayUrlWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param tmpReq - GetOrgLiveListRequest
     * @param headers - GetOrgLiveListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns GetOrgLiveListResponse
     */
    async getOrgLiveListWithOptions(tmpReq, headers, runtime) {
        tea_util_1.default.validateModel(tmpReq);
        let request = new GetOrgLiveListShrinkRequest({});
        openapi_util_1.default.convert(tmpReq, request);
        if (!tea_util_1.default.isUnset(tmpReq.requestBody)) {
            request.requestBodyShrink = openapi_util_1.default.arrayToStringWithSpecifiedStyle(tmpReq.requestBody, "requestBody", "json");
        }
        let query = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            query["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.requestBodyShrink)) {
            query["requestBody"] = request.requestBodyShrink;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetOrgLiveList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/organizations/liveLists/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetOrgLiveListResponse({}));
    }
    /**
     * 
     *
     * @param request - GetOrgLiveListRequest
     * @returns GetOrgLiveListResponse
     */
    async getOrgLiveList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetOrgLiveListHeaders({});
        return await this.getOrgLiveListWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - GetUserAllLiveListRequest
     * @param headers - GetUserAllLiveListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns GetUserAllLiveListResponse
     */
    async getUserAllLiveListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.statuses)) {
            body["statuses"] = request.statuses;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "GetUserAllLiveList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/users/allLiveInfos/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetUserAllLiveListResponse({}));
    }
    /**
     * 
     *
     * @param request - GetUserAllLiveListRequest
     * @returns GetUserAllLiveListResponse
     */
    async getUserAllLiveList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUserAllLiveListHeaders({});
        return await this.getUserAllLiveListWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - GetUserCreateLiveListRequest
     * @param headers - GetUserCreateLiveListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns GetUserCreateLiveListResponse
     */
    async getUserCreateLiveListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.statuses)) {
            body["statuses"] = request.statuses;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "GetUserCreateLiveList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/users/createLiveInfos/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetUserCreateLiveListResponse({}));
    }
    /**
     * 
     *
     * @param request - GetUserCreateLiveListRequest
     * @returns GetUserCreateLiveListResponse
     */
    async getUserCreateLiveList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUserCreateLiveListHeaders({});
        return await this.getUserCreateLiveListWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - GetUserWatchLiveListRequest
     * @param headers - GetUserWatchLiveListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns GetUserWatchLiveListResponse
     */
    async getUserWatchLiveListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.filterType)) {
            query["filterType"] = request.filterType;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetUserWatchLiveList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/users/watchRecords`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetUserWatchLiveListResponse({}));
    }
    /**
     * 
     *
     * @param request - GetUserWatchLiveListRequest
     * @returns GetUserWatchLiveListResponse
     */
    async getUserWatchLiveList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUserWatchLiveListHeaders({});
        return await this.getUserWatchLiveListWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param tmpReq - ModifyFeedWhiteListRequest
     * @param headers - ModifyFeedWhiteListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns ModifyFeedWhiteListResponse
     */
    async modifyFeedWhiteListWithOptions(feedId, tmpReq, headers, runtime) {
        tea_util_1.default.validateModel(tmpReq);
        let request = new ModifyFeedWhiteListShrinkRequest({});
        openapi_util_1.default.convert(tmpReq, request);
        if (!tea_util_1.default.isUnset(tmpReq.modifyUserList)) {
            request.modifyUserListShrink = openapi_util_1.default.arrayToStringWithSpecifiedStyle(tmpReq.modifyUserList, "modifyUserList", "json");
        }
        let query = {};
        if (!tea_util_1.default.isUnset(request.action)) {
            query["action"] = request.action;
        }
        if (!tea_util_1.default.isUnset(request.modifyUserListShrink)) {
            query["modifyUserList"] = request.modifyUserListShrink;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ModifyFeedWhiteList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/openFeeds/${feedId}/whiteList`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ModifyFeedWhiteListResponse({}));
    }
    /**
     * 
     *
     * @param request - ModifyFeedWhiteListRequest
     * @returns ModifyFeedWhiteListResponse
     */
    async modifyFeedWhiteList(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ModifyFeedWhiteListHeaders({});
        return await this.modifyFeedWhiteListWithOptions(feedId, request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - QueryFeedWhiteListRequest
     * @param headers - QueryFeedWhiteListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns QueryFeedWhiteListResponse
     */
    async queryFeedWhiteListWithOptions(feedId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryFeedWhiteList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/openFeeds/${feedId}/whiteList`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryFeedWhiteListResponse({}));
    }
    /**
     * 
     *
     * @param request - QueryFeedWhiteListRequest
     * @returns QueryFeedWhiteListResponse
     */
    async queryFeedWhiteList(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryFeedWhiteListHeaders({});
        return await this.queryFeedWhiteListWithOptions(feedId, request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - QueryLiveInfoRequest
     * @param headers - QueryLiveInfoHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns QueryLiveInfoResponse
     */
    async queryLiveInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryLiveInfo",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/lives`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryLiveInfoResponse({}));
    }
    /**
     * 
     *
     * @param request - QueryLiveInfoRequest
     * @returns QueryLiveInfoResponse
     */
    async queryLiveInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryLiveInfoHeaders({});
        return await this.queryLiveInfoWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - QueryLiveInteractionPluginRequest
     * @param headers - QueryLiveInteractionPluginHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns QueryLiveInteractionPluginResponse
     */
    async queryLiveInteractionPluginWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.pluginId)) {
            query["pluginId"] = request.pluginId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryLiveInteractionPlugin",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/interactionPlugins`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryLiveInteractionPluginResponse({}));
    }
    /**
     * 
     *
     * @param request - QueryLiveInteractionPluginRequest
     * @returns QueryLiveInteractionPluginResponse
     */
    async queryLiveInteractionPlugin(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryLiveInteractionPluginHeaders({});
        return await this.queryLiveInteractionPluginWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - QueryLiveWatchDetailRequest
     * @param headers - QueryLiveWatchDetailHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns QueryLiveWatchDetailResponse
     */
    async queryLiveWatchDetailWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryLiveWatchDetail",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/lives/watchDetails`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryLiveWatchDetailResponse({}));
    }
    /**
     * 
     *
     * @param request - QueryLiveWatchDetailRequest
     * @returns QueryLiveWatchDetailResponse
     */
    async queryLiveWatchDetail(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryLiveWatchDetailHeaders({});
        return await this.queryLiveWatchDetailWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - QueryLiveWatchUserListRequest
     * @param headers - QueryLiveWatchUserListHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns QueryLiveWatchUserListResponse
     */
    async queryLiveWatchUserListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryLiveWatchUserList",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/lives/watchUsers`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryLiveWatchUserListResponse({}));
    }
    /**
     * 
     *
     * @param request - QueryLiveWatchUserListRequest
     * @returns QueryLiveWatchUserListResponse
     */
    async queryLiveWatchUserList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryLiveWatchUserListHeaders({});
        return await this.queryLiveWatchUserListWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param tmpReq - QuerySubscribeStatusRequest
     * @param headers - QuerySubscribeStatusHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns QuerySubscribeStatusResponse
     */
    async querySubscribeStatusWithOptions(tmpReq, headers, runtime) {
        tea_util_1.default.validateModel(tmpReq);
        let request = new QuerySubscribeStatusShrinkRequest({});
        openapi_util_1.default.convert(tmpReq, request);
        if (!tea_util_1.default.isUnset(tmpReq.body)) {
            request.bodyShrink = openapi_util_1.default.arrayToStringWithSpecifiedStyle(tmpReq.body, "body", "json");
        }
        let query = {};
        if (!tea_util_1.default.isUnset(request.bodyShrink)) {
            query["body"] = request.bodyShrink;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QuerySubscribeStatus",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/subscribeStatuses/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QuerySubscribeStatusResponse({}));
    }
    /**
     * 
     *
     * @param request - QuerySubscribeStatusRequest
     * @returns QuerySubscribeStatusResponse
     */
    async querySubscribeStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QuerySubscribeStatusHeaders({});
        return await this.querySubscribeStatusWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - SendLiveInteractionPluginEffectsMsgRequest
     * @param headers - SendLiveInteractionPluginEffectsMsgHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns SendLiveInteractionPluginEffectsMsgResponse
     */
    async sendLiveInteractionPluginEffectsMsgWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.pluginId)) {
            query["pluginId"] = request.pluginId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.count)) {
            body["count"] = request.count;
        }
        if (!tea_util_1.default.isUnset(request.lottieFileUrl)) {
            body["lottieFileUrl"] = request.lottieFileUrl;
        }
        if (!tea_util_1.default.isUnset(request.msgIconUrl)) {
            body["msgIconUrl"] = request.msgIconUrl;
        }
        if (!tea_util_1.default.isUnset(request.msgText)) {
            body["msgText"] = request.msgText;
        }
        if (!tea_util_1.default.isUnset(request.pluginSubId)) {
            body["pluginSubId"] = request.pluginSubId;
        }
        if (!tea_util_1.default.isUnset(request.senderUnionId)) {
            body["senderUnionId"] = request.senderUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendLiveInteractionPluginEffectsMsg",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/interactionPlugins/effectMessages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendLiveInteractionPluginEffectsMsgResponse({}));
    }
    /**
     * 
     *
     * @param request - SendLiveInteractionPluginEffectsMsgRequest
     * @returns SendLiveInteractionPluginEffectsMsgResponse
     */
    async sendLiveInteractionPluginEffectsMsg(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendLiveInteractionPluginEffectsMsgHeaders({});
        return await this.sendLiveInteractionPluginEffectsMsgWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param tmpReq - SendLivePluginUserActionMsgRequest
     * @param headers - SendLivePluginUserActionMsgHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns SendLivePluginUserActionMsgResponse
     */
    async sendLivePluginUserActionMsgWithOptions(tmpReq, headers, runtime) {
        tea_util_1.default.validateModel(tmpReq);
        let request = new SendLivePluginUserActionMsgShrinkRequest({});
        openapi_util_1.default.convert(tmpReq, request);
        if (!tea_util_1.default.isUnset(tmpReq.pluginEffectsMessage)) {
            request.pluginEffectsMessageShrink = openapi_util_1.default.arrayToStringWithSpecifiedStyle(tmpReq.pluginEffectsMessage, "pluginEffectsMessage", "json");
        }
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.pluginEffectsMessageShrink)) {
            query["pluginEffectsMessage"] = request.pluginEffectsMessageShrink;
        }
        if (!tea_util_1.default.isUnset(request.pluginId)) {
            query["pluginId"] = request.pluginId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "SendLivePluginUserActionMsg",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/interactionPlugins/actionMessages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendLivePluginUserActionMsgResponse({}));
    }
    /**
     * 
     *
     * @param request - SendLivePluginUserActionMsgRequest
     * @returns SendLivePluginUserActionMsgResponse
     */
    async sendLivePluginUserActionMsg(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendLivePluginUserActionMsgHeaders({});
        return await this.sendLivePluginUserActionMsgWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - StartCloudFeedRequest
     * @param headers - StartCloudFeedHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns StartCloudFeedResponse
     */
    async startCloudFeedWithOptions(feedId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "StartCloudFeed",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/cloudFeeds/${feedId}/start`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new StartCloudFeedResponse({}));
    }
    /**
     * 
     *
     * @param request - StartCloudFeedRequest
     * @returns StartCloudFeedResponse
     */
    async startCloudFeed(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new StartCloudFeedHeaders({});
        return await this.startCloudFeedWithOptions(feedId, request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - StopCloudFeedRequest
     * @param headers - StopCloudFeedHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns StopCloudFeedResponse
     */
    async stopCloudFeedWithOptions(feedId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "StopCloudFeed",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/cloudFeeds/${feedId}/stop`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new StopCloudFeedResponse({}));
    }
    /**
     * 
     *
     * @param request - StopCloudFeedRequest
     * @returns StopCloudFeedResponse
     */
    async stopCloudFeed(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new StopCloudFeedHeaders({});
        return await this.stopCloudFeedWithOptions(feedId, request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - SubscribeLiveRequest
     * @param headers - SubscribeLiveHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns SubscribeLiveResponse
     */
    async subscribeLiveWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.subscribe)) {
            query["subscribe"] = request.subscribe;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "SubscribeLive",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/lives/subscribe`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SubscribeLiveResponse({}));
    }
    /**
     * 
     *
     * @param request - SubscribeLiveRequest
     * @returns SubscribeLiveResponse
     */
    async subscribeLive(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SubscribeLiveHeaders({});
        return await this.subscribeLiveWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - UpdateLiveRequest
     * @param headers - UpdateLiveHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns UpdateLiveResponse
     */
    async updateLiveWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.coverUrl)) {
            body["coverUrl"] = request.coverUrl;
        }
        if (!tea_util_1.default.isUnset(request.introduction)) {
            body["introduction"] = request.introduction;
        }
        if (!tea_util_1.default.isUnset(request.liveId)) {
            body["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.preEndTime)) {
            body["preEndTime"] = request.preEndTime;
        }
        if (!tea_util_1.default.isUnset(request.preStartTime)) {
            body["preStartTime"] = request.preStartTime;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateLive",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/lives`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateLiveResponse({}));
    }
    /**
     * 
     *
     * @param request - UpdateLiveRequest
     * @returns UpdateLiveResponse
     */
    async updateLive(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateLiveHeaders({});
        return await this.updateLiveWithOptions(request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - UpdateLiveFeedRequest
     * @param headers - UpdateLiveFeedHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns UpdateLiveFeedResponse
     */
    async updateLiveFeedWithOptions(feedId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.coverUrl)) {
            query["coverUrl"] = request.coverUrl;
        }
        if (!tea_util_1.default.isUnset(request.introduction)) {
            query["introduction"] = request.introduction;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            query["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            query["title"] = request.title;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "UpdateLiveFeed",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/openFeeds/${feedId}`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateLiveFeedResponse({}));
    }
    /**
     * 
     *
     * @param request - UpdateLiveFeedRequest
     * @returns UpdateLiveFeedResponse
     */
    async updateLiveFeed(feedId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateLiveFeedHeaders({});
        return await this.updateLiveFeedWithOptions(feedId, request, headers, runtime);
    }
    /**
     * 
     *
     * @param request - UpdateLiveInteractionPluginRequest
     * @param headers - UpdateLiveInteractionPluginHeaders
     * @param runtime - runtime options for this request RuntimeOptions
     * @returns UpdateLiveInteractionPluginResponse
     */
    async updateLiveInteractionPluginWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.liveId)) {
            query["liveId"] = request.liveId;
        }
        if (!tea_util_1.default.isUnset(request.pluginId)) {
            query["pluginId"] = request.pluginId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.anchorJumpUrl)) {
            body["anchorJumpUrl"] = request.anchorJumpUrl;
        }
        if (!tea_util_1.default.isUnset(request.pluginIconUrl)) {
            body["pluginIconUrl"] = request.pluginIconUrl;
        }
        if (!tea_util_1.default.isUnset(request.pluginName)) {
            body["pluginName"] = request.pluginName;
        }
        if (!tea_util_1.default.isUnset(request.pluginNameEn)) {
            body["pluginNameEn"] = request.pluginNameEn;
        }
        if (!tea_util_1.default.isUnset(request.viewerJumpUrl)) {
            body["viewerJumpUrl"] = request.viewerJumpUrl;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateLiveInteractionPlugin",
            version: "live_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/live/interactionPlugins`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateLiveInteractionPluginResponse({}));
    }
    /**
     * 
     *
     * @param request - UpdateLiveInteractionPluginRequest
     * @returns UpdateLiveInteractionPluginResponse
     */
    async updateLiveInteractionPlugin(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateLiveInteractionPluginHeaders({});
        return await this.updateLiveInteractionPluginWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
